src/components/NewRuntimeModal.js

```diff
@@ -751,51 +770,138 @@ export class NewRuntimeModalBase extends Component {
-            div({ style: { gridColumnEnd: 'span 2' } }),
-          h(IdContainer, [
-            id => div({ style: { gridColumnEnd: 'span 6' } }, [
-              label({ htmlFor: id, style: styles.label }, ['Startup script']),
-              div({ style: { marginTop: '0.5rem' } }, [
-                h(TextInput, {
-                  id,
-                  placeholder: 'URI',
-                  value: jupyterUserScriptUri,
-                  onChange: v => this.setState({ jupyterUserScriptUri: v })
-                })
+        div([
+          div({ style: { ...gridStyle, gridTemplateColumns: `0.25fr 4.5rem 1fr 5.5rem 1fr 5.5rem` } }, [
+            // CPU & Memory Selection
+            h(Fragment, [
+              h(IdContainer, [
+                id => h(Fragment, [
+                  label({ htmlFor: id, style: styles.label }, ['CPUs']),
+                  div([
+                    h(Select, {
+                      id,
+                      isSearchable: false,
+                      value: currentNumCpus,
+                      onChange: option => this.setState({ masterMachineType: _.find({ cpu: option.value }, validMachineTypes)?.name || mainMachineType }),
+                      options: _.flow(_.map('cpu'), _.union([currentNumCpus]), _.sortBy(_.identity))(validMachineTypes)
+                    })
+                  ])
+                ])
+              ]),
+              h(IdContainer, [
+                id => h(Fragment, [
+                  label({ htmlFor: id, style: styles.label }, ['Memory (GB)']),
+                  div([
+                    h(Select, {
+                      id,
+                      isSearchable: false,
+                      value: currentMemory,
+                      onChange: option => this.setState({ masterMachineType: _.find({ cpu: currentNumCpus, memory: option.value }, validMachineTypes)?.name || mainMachineType }),
+                      options: _.flow(_.filter({ cpu: currentNumCpus }), _.map('memory'), _.union([currentMemory]), _.sortBy(_.identity))(validMachineTypes)
+                    })
+                  ])
+                ])
+              ])
+            ]),
+            // Disk Selection
+            !isPersistentDisk ?
+              h(DiskSelector, { value: masterDiskSize, onChange: v => this.setState({ masterDiskSize: v }) }) :
+              div({ style: { gridColumnEnd: 'span 2' } })
+          ]),
+          // GPU Enabling
+          !sparkMode && div({ style: { gridColumnEnd: 'span 6', marginTop: '1.25rem' } }, [
+            h(LabeledCheckbox, {
+              checked: gpuEnabled,
+              onChange: v => this.setState({ gpuEnabled: v || hasGpu })
+            }, [span({ style: { marginLeft: '0.5rem', ...styles.label, verticalAlign: 'top' } }, ['Enable GPUs ', versionTag('Beta', { color: colors.primary(1.5), backgroundColor: 'white', border: `1px solid ${colors.primary(1.5)}` })]),
```

Comment:
Just curious: is there a particular reason this is `Beta`? Is there criteria for moving it out of Beta? Just thinking that while there may be evolution of other ML functionality, GPUs specifically are pretty much done.

Reply:
Good question. I think I'd feel more comfortable with removing `Beta` once we
1. gain more confidence that our revamped container image stack is more proven
2. have a better idea of how much the Google GPU quota issue will affect user experience, and what we can do about it
3. get good feedback from beta testers that it's usable for them

What do you think?

Reply:
That all sounds good to me 👍 , thanks

Reply:
This is an extremely long line, could you break it up? As a guideline, try to avoid lines much over ~120 characters.